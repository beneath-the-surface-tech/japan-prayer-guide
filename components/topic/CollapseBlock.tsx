import React, { ReactNode } from "react"
import { Collapse, Container } from "react-bootstrap"
import Button from "react-bootstrap/Button"
import { RiArrowDownSLine, RiArrowUpSLine } from "react-icons/ri"

interface feedbackProps {
    title: string
    children: ReactNode
    startOpened?: boolean
}

const CollapseBlock = ({ title, children, startOpened = false }: feedbackProps) => {
    const [openCollapse, setOpenCollapse] = React.useState(startOpened)

    return (
        <Container>
            <hr />
            <Button
                className="collapse-label fs-2 fw-semibold lh-sm ps-0"
                onClick={() => setOpenCollapse(!openCollapse)}
                aria-controls="collapse-block"
                aria-expanded={openCollapse}
            >
                {title} {openCollapse ? <RiArrowUpSLine /> : <RiArrowDownSLine />}
            </Button>
            <Collapse in={openCollapse}>
                <Container id="collapse-block">{children}</Container>
            </Collapse>
        </Container>
    )
}

export default CollapseBlock
