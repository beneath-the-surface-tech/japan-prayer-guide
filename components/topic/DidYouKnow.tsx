import React from "react";
import { Container } from "react-bootstrap";
import { IconContext } from "react-icons";
import { BsLightbulb } from "react-icons/bs";

interface dykProps {
    text: string
    title: string
}

const DidYouKnow = ({text, title}:dykProps) => {
    return (
        <Container data-testid={"dyk-container"} className="d-flex justify-content-center">
            <div className="did-you-know p-4 shadow">
                <div className="d-flex align-items-center">
                    <IconContext.Provider value={{size: '25px'}}>
                        <BsLightbulb/>
                    </IconContext.Provider>
                    <h1 className="fw-bold fs-5 my-0 mx-2">{title}</h1>
                </div>
                <div data-testid={"dyk-text"} className="fs-5 my-3">
                    {text}
                </div>
            </div>
        </Container>
    )
}

export default DidYouKnow