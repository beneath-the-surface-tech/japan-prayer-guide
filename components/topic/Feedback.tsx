import { Trans, useTranslation } from "next-i18next"
import React from "react"
import Button from "react-bootstrap/Button"

const Feedback = () => {
    const { t } = useTranslation("common")
    return (
        <section
            data-testid={"feedback-sect"}
            className="text-center m-0 p-4 w-100"
            style={{ backgroundColor: "#EAF4FF" }}
        >
            <h1 data-testid={"feedback-title"} className="text-primary fs-2">
                <Trans t={t} i18nKey="feedback.title" />
            </h1>
            <Button className="bg-secondary text-white border-0 p-2 w-100 mt-4" style={{ maxWidth: "300px" }}>
                <p data-testid={"feedback-button"} className="m-0">
                    <Trans t={t} i18nKey="feedback.button" />
                </p>
            </Button>
        </section>
    )
}

export default Feedback
