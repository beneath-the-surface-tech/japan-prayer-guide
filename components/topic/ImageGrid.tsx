import React from "react";
import { Container, Carousel } from "react-bootstrap";
import Row from "react-bootstrap/Row";
import Col from "react-bootstrap/Col";

const ImageGroup = () => {
    return (
        <Container className="w-100 d-flex justify-content-center">
            <ImageGrid/>
            <ImageCarousel/>
        </Container>
    )
}

const ImageGrid = () => {
    return (
        <div className="w-100 d-flex flex-column gap-3 d-none d-lg-flex my-3">
            <Row className="">
                <Col>
                    <div id="about-landing-image" className=""></div>
                </Col>
                <Col>
                    <div id="about-landing-image" className=""></div>
                </Col>
            </Row>
            <Row>
                <Col>
                    <div id="about-landing-image" className=""></div>
                </Col>
                <Col>
                    <div id="about-landing-image" className=""></div>
                </Col>
            </Row>
        </div>
    )
}

const ImageCarousel = () => {
    return (
        <Carousel id="image-carousel" className="w-100 d-lg-none" interval={null}>
            <Carousel.Item>
                <div id="about-landing-image" className=""></div>
            </Carousel.Item>
            <Carousel.Item>
                <div id="about-landing-image" className=""></div>
            </Carousel.Item>
            <Carousel.Item>
                <div id="about-landing-image" className=""></div>
            </Carousel.Item>
            <Carousel.Item>
                <div id="about-landing-image" className=""></div>
            </Carousel.Item>
        </Carousel>
    )
}

export default ImageGroup