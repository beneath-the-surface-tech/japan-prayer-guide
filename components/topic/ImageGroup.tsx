import React from "react"
import { Container, Carousel } from "react-bootstrap"
import Row from "react-bootstrap/Row"
import Col from "react-bootstrap/Col"

// TODO: Should accept an array of image links which are displayed here

const ImageGroup = () => {
  return (
    <Container className="w-100 d-flex justify-content-center">
      <ImageGrid />
      <ImageCarousel />
    </Container>
  )
}

const ImageGrid = () => {
  return (
    <div className="w-100 d-flex flex-column gap-3 d-none d-md-flex my-3">
      <Row>
        <Col>
          <div className="placeholder-image"></div>
        </Col>
        <Col>
          <div className="placeholder-image"></div>
        </Col>
      </Row>
      <Row>
        <Col>
          <div className="placeholder-image"></div>
        </Col>
        <Col>
          <div className="placeholder-image"></div>
        </Col>
      </Row>
    </div>
  )
}

const ImageCarousel = () => {
  return (
    <Carousel id="image-carousel" className="w-100 d-md-none" interval={null}>
      <Carousel.Item>
        <div className="placeholder-image"></div>
      </Carousel.Item>
      <Carousel.Item>
        <div className="placeholder-image"></div>
      </Carousel.Item>
      <Carousel.Item>
        <div className="placeholder-image"></div>
      </Carousel.Item>
      <Carousel.Item>
        <div className="placeholder-image"></div>
      </Carousel.Item>
    </Carousel>
  )
}

export default ImageGroup
