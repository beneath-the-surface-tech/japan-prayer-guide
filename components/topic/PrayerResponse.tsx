import { TFunction, Trans, useTranslation } from "next-i18next"
import React from "react"
import { Col, Container, FloatingLabel, Form, Row } from "react-bootstrap"
import Button from "react-bootstrap/Button"
import PrayerPoints, { PrayerDisplayStyle } from "../common/PrayerPoints"

const PrayerResponse = ({ topicTrans }: { topicTrans: TFunction }) => {
    const { t: common } = useTranslation("common")

    const textPlaceholder: string = common("response.feedbackTextPlaceholder", "")
    const emailPlaceholder: string = common("response.feedbackEmailPlaceholder", "")

    const topicVideoSrc: string = topicTrans("videoSrc", "")
    const couldLoadVideo: boolean = topicVideoSrc !== ""

    return (
        <Container fluid id="prayer-response" data-testid="prayer-response" className="pb-5">
            <Container>
                <h2 className="fw-bold text-primary">
                    <Trans t={common} i18nKey="response.title" />
                </h2>
                <p>
                    <Trans t={common} i18nKey="response.blurb0" />
                </p>
                <p>
                    <Trans t={common} i18nKey="response.blurb1" />
                </p>

                <h3 className="fw-bold fs-2">
                    <Trans t={common} i18nKey="response.recordingLabel" />
                </h3>
                <div className="mb-5 video-embed">
                    {couldLoadVideo ? (
                        <iframe
                            className="d-flex align-content-start"
                            src={topicVideoSrc}
                            allow="autoplay; fullscreen"
                            style={{
                                position: "absolute",
                                top: "0",
                                left: "0",
                                width: "100%",
                                height: "100%",
                                border: "0px",
                            }}
                            title="Prayer recording"
                        />
                    ) : (
                        <div>
                            <Trans t={common} i18nKey="errors.videoLoadError" />
                        </div>
                    )}
                </div>
                <Row>
                    <Col lg={6} md={12} className="px-0">
                        <PrayerPoints topicTrans={topicTrans} displayStyle={PrayerDisplayStyle.TopicBottom} />
                    </Col>
                    <Col lg={6} md={12}>
                        <Container data-testid="feedback" fluid className="px-0">
                            <h3 className="fw-bold mt-0">
                                <Trans t={common} i18nKey="response.feedbackLabel" />
                            </h3>
                            <Form action="https://formbold.com/s/67Gj1" method="POST">
                                <Form.Group className="mb-2">
                                    <FloatingLabel
                                        controlId="floatingTextArea"
                                        label={textPlaceholder}
                                        className="text-muted"
                                    >
                                        <Form.Control
                                            as="textarea"
                                            name="message"
                                            placeholder={textPlaceholder}
                                            style={{ height: "100px" }}
                                        />
                                    </FloatingLabel>
                                </Form.Group>
                                <Form.Group className="mb-3">
                                    <Form.Control type="email" placeholder={emailPlaceholder} />
                                </Form.Group>
                                <Button variant="outline-primary" className="fw-bold px-4 py-2" type="submit">
                                    <Trans t={common} i18nKey="response.feedbackSubmit" />
                                </Button>
                            </Form>
                        </Container>
                    </Col>
                </Row>
            </Container>
        </Container>
    )
}

export default PrayerResponse
