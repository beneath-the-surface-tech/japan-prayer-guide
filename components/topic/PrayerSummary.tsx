import React from "react";
import { Container, Card } from "react-bootstrap";
import {FaPrayingHands} from 'react-icons/fa'
import { IconContext } from "react-icons/lib";
import { BsDownload } from "react-icons/bs";

interface Props {
    prayerPoints: string[]
}

export default function PrayerSummary({prayerPoints}:Props) {
    return (
        <Container className="d-flex justify-content-center px-6">
            <Card className="my-4 shadow" style={{maxWidth: '400px'}}>
                <Card.Body>
                    <Card.Text className="text-primary px-2 py-3 fs-3 fst-italic border-bottom border-grey d-flex justify-content-between align-items-center">
                        Please pray for:
                        <IconContext.Provider value={{size:'30px'}}>
                            <BsDownload className="text-secondary fw-bold" style={{cursor: 'pointer'}}>
                            </BsDownload>
                        </IconContext.Provider>
                    </Card.Text>
                    <ul className="fs-5">
                        {prayerPoints.map((point:string) => 
                            <li key={point} className="my-3">{point}</li>
                        )}
                    </ul>
                </Card.Body>
            </Card>
        </Container>
    )
}