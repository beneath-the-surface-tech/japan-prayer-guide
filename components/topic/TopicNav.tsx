import { TFunction, Trans } from "next-i18next"
import { Dispatch, SetStateAction, useEffect, useRef, useState } from "react"
import { Container } from "react-bootstrap"

interface navProps {
    selected: string
    setSelected: Dispatch<SetStateAction<string>>
    topicTrans: TFunction
}

export const TopicNav = ({ selected, setSelected, topicTrans }: navProps) => {
    const bannerRef = useRef(null)
    const [show, setShow] = useState(false)
    const labels: string[] = topicTrans("topicNav.labels", { returnObjects: true })

    useEffect(() => {
        const handleScroll = () => {
            const bannerElement: any = bannerRef.current
            if (bannerElement) {
                const { top } = bannerElement.getBoundingClientRect()
                if (top === 0) {
                    setShow(true)
                } else {
                    setShow(false)
                }
            }
        }
        window.addEventListener("scroll", handleScroll)
        return () => {
            window.removeEventListener("scroll", handleScroll)
        }
    }, [])

    return (
        <Container
            style={{ maxWidth: "none" }}
            ref={bannerRef}
            data-testid={"topic-nav-container"}
            className="w-100 d-flex justify-content-center flex-column sticky-top p-0"
        >
            <div id="overview-nav" className="w-100" style={show ? {} : { height: "0px" }}></div>
            <div
                data-testid={"topic-nav-links"}
                className="pt-4 pb-0 w-100 border-bottom d-flex justify-content-around bg-white"
            >
                <a
                    href="#topic-about"
                    className={
                        "px-3 my-0 text-decoration-none fs-4" +
                        (selected === "About" ? " border-bottom border-secondary border-3 text-secondary" : "")
                    }
                    onClick={() => setSelected("About")}
                    data-testid={"topic-nav-link"}
                >
                    <Trans>{labels[0]}</Trans>
                </a>
                <a
                    href="#topic-prayer-topicBottom"
                    className={
                        "px-3 my-0 text-decoration-none fs-4" +
                        (selected === "Prayer" ? " border-bottom border-secondary border-3 text-secondary" : "")
                    }
                    onClick={() => setSelected("Prayer")}
                >
                    <Trans>{labels[1]}</Trans>
                </a>
                <a
                    href="#topic-downloads"
                    className={
                        "px-3 my-0 text-decoration-none fs-4" +
                        (selected === "Downloads" ? " border-bottom border-secondary border-3 text-secondary" : "")
                    }
                    onClick={() => setSelected("Downloads")}
                >
                    <Trans>{labels[2]}</Trans>
                </a>
            </div>
        </Container>
    )
}
