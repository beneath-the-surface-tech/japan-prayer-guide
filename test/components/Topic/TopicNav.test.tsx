import { render, screen } from "@testing-library/react"
import { TopicNav } from "../../../components/topic/TopicNav"

jest.mock("next-i18next", () => ({
    // this mock makes sure any components using the translate hook can use it without a warning being shown
    useTranslation: () => {
        return {
            t: (str) => str,
            i18n: {
                changeLanguage: () => new Promise(() => {}),
            },
        }
    },
    Trans: (props: any) => {
        if (props.children) {
            // mock for <Trans>{value}</Trans>
            return props.children
        } else if (props.i18nKey) {
            // mock for <Trans t={t} i18nKey={value} />
            return props.i18nKey
        }
        return props
    },
}))

describe("Topic Nav", () => {
    test("Snapshot test", () => {
        // This test is to check against unintended changes.
        // If the change is intentional you can update the snapshot with `jest --updateSnapshot`

        const labels = ["About", "Prayer", "Download"]
        const t = jest.fn()
        t.mockReturnValueOnce(labels)

        const component = render(<TopicNav selected={"About"} setSelected={() => {}} topicTrans={t} />)
        expect(component).toMatchSnapshot()
    })

    test("Renders a section with the right text inside", () => {
        const labels = ["About this Topic", "Prayer Points", "Downloadables"]
        const t = jest.fn()
        t.mockReturnValueOnce(labels)

        render(<TopicNav selected={"About"} setSelected={() => {}} topicTrans={t} />)
        const topicNavCont = screen.getByTestId("topic-nav-container")
        const topicNavLinks = screen.getByTestId("topic-nav-links")
        const topicNavLink = screen.getByTestId("topic-nav-link")

        expect(topicNavCont).toHaveClass("d-flex", "container", "w-100")

        expect(topicNavLinks).toHaveClass("pt-4", "pb-0", "w-100")
        expect(topicNavLinks.children.length).toBe(3)

        expect(topicNavLink).toHaveTextContent("About this Topic")
        expect(topicNavLink).toHaveClass("px-3", "my-0", "text-decoration-none")
    })
})
