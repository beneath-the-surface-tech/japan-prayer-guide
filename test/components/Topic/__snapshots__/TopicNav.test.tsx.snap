// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Topic Nav Snapshot test 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="w-100 d-flex justify-content-center flex-column sticky-top p-0 container"
        data-testid="topic-nav-container"
        style="max-width: none;"
      >
        <div
          class="w-100"
          id="overview-nav"
          style="height: 0px;"
        />
        <div
          class="pt-4 pb-0 w-100 border-bottom d-flex justify-content-around bg-white"
          data-testid="topic-nav-links"
        >
          <a
            class="px-3 my-0 text-decoration-none fs-4 border-bottom border-secondary border-3 text-secondary"
            data-testid="topic-nav-link"
            href="#topic-about"
          >
            About
          </a>
          <a
            class="px-3 my-0 text-decoration-none fs-4"
            href="#topic-prayer-topicBottom"
          >
            Prayer
          </a>
          <a
            class="px-3 my-0 text-decoration-none fs-4"
            href="#topic-downloads"
          >
            Download
          </a>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="w-100 d-flex justify-content-center flex-column sticky-top p-0 container"
      data-testid="topic-nav-container"
      style="max-width: none;"
    >
      <div
        class="w-100"
        id="overview-nav"
        style="height: 0px;"
      />
      <div
        class="pt-4 pb-0 w-100 border-bottom d-flex justify-content-around bg-white"
        data-testid="topic-nav-links"
      >
        <a
          class="px-3 my-0 text-decoration-none fs-4 border-bottom border-secondary border-3 text-secondary"
          data-testid="topic-nav-link"
          href="#topic-about"
        >
          About
        </a>
        <a
          class="px-3 my-0 text-decoration-none fs-4"
          href="#topic-prayer-topicBottom"
        >
          Prayer
        </a>
        <a
          class="px-3 my-0 text-decoration-none fs-4"
          href="#topic-downloads"
        >
          Download
        </a>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
